src/Common/Core/Impl/OS/Win32Process.cs

```diff
@@ -0,0 +1,111 @@
+            lpDesktop = "<windowsta>\<desktop>"; the system will try to start the process into that windowstation and desktop.
+            lpDesktop = NULL; the system will try to use the same windowstation and desktop as the calling process if the system is associated with the interactive windowstation.
+            lpDesktop = <somevalue>; the system will create a new windowstation and desktop that you cannot see.
+            lpDesktop = ""; it will either create a new windowstation and desktop that you cannot see, or if one has been created by means of a prior call by using the same access token, the existing windowstation and desktop will be used.
+            */
+            si.lpDesktop = "";
+
+            IntPtr stdinRead, stdinWrite, stdoutRead, stdoutWrite, stderrorRead, stderrorWrite;
+
+            SECURITY_ATTRIBUTES sa = default(SECURITY_ATTRIBUTES);
+            sa.nLength = Marshal.SizeOf(sa);
+            sa.lpSecurityDescriptor = IntPtr.Zero;
+            sa.bInheritHandle = true;
+
+            if (!CreatePipe(out stdinRead, out stdinWrite, ref sa, 0)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            if (!SetHandleInformation(stdinWrite, HANDLE_FLAGS.INHERIT, HANDLE_FLAGS.None)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            if (!CreatePipe(out stdoutRead, out stdoutWrite, ref sa, 0)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            if (!SetHandleInformation(stdoutRead, HANDLE_FLAGS.INHERIT, HANDLE_FLAGS.None)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            if (!CreatePipe(out stderrorRead, out stderrorWrite, ref sa, 0)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            if (!SetHandleInformation(stderrorRead, HANDLE_FLAGS.INHERIT, HANDLE_FLAGS.None)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
+            }
+
+            si.dwFlags = STARTF_USESTDHANDLES;
+            si.hStdInput = stdinRead;
+            si.hStdOutput = stdoutWrite;
+            si.hStdError = stderrorWrite;
+
+            SECURITY_ATTRIBUTES processAttr = CreateSecurityAttributes();
+            SECURITY_ATTRIBUTES threadAttr = CreateSecurityAttributes();
+
+            PROCESS_INFORMATION pi;
+            if (!CreateProcessAsUser(
+                winIdentity.Token, applicationName, commandLine, ref processAttr, ref threadAttr, true,
+                (uint)(CREATE_PROCESS_FLAGS.CREATE_UNICODE_ENVIRONMENT | CREATE_PROCESS_FLAGS.CREATE_NO_WINDOW),
+                environment.NativeEnvironmentBlock,
+                workingDirectory,
+                ref si,
+                out pi)) {
+                throw new Win32Exception(Marshal.GetLastWin32Error());
```

Comment:
You may want to free `processAttr` and `threadAttr` if `CreateProcessAsUser` fails